src/middlewared/middlewared/etc_files/hostname.py

```diff
@@ -7,4 +7,8 @@
 
 async def render(service, middleware):
     config = await middleware.call("network.configuration.config")
-    await run(["hostname", f'{config["hostname_local"]}.{config["domain"]}'])
+    hostname = f"{config['hostname_local']}.{config['domain']}"
```

Comment:
As discussed in slack, `/etc/hostname` should be the short-hand name (`hostname_local`) and `/etc/resolv.conf` has the domain name portion.